stdlib/public/runtime/SwiftRT-ELF.cpp

```diff
@@ -34,41 +34,53 @@ DECLARE_SWIFT_SECTION(swift5_fieldmd)
 DECLARE_SWIFT_SECTION(swift5_assocty)
 DECLARE_SWIFT_SECTION(swift5_replace)
 DECLARE_SWIFT_SECTION(swift5_replac2)
+DECLARE_SWIFT_SECTION(swift5_builtin)
+DECLARE_SWIFT_SECTION(swift5_capture)
 }
 
 #undef DECLARE_SWIFT_SECTION
 
-namespace {
-static swift::MetadataSections sections{};
-}
-
-__attribute__((__constructor__))
-static void swift_image_constructor() {
 #define SWIFT_SECTION_RANGE(name)                                              \
   { reinterpret_cast<uintptr_t>(&__start_##name),                              \
     static_cast<uintptr_t>(&__stop_##name - &__start_##name) }
 
-  sections = {
-      swift::CurrentSectionMetadataVersion,
-      0,
+static swift::MetadataSections sections{
+    swift::CurrentSectionMetadataVersion,
+    0,
 
-      nullptr,
-      nullptr,
+    nullptr,
+    nullptr,
 
-      SWIFT_SECTION_RANGE(swift5_protocols),
-      SWIFT_SECTION_RANGE(swift5_protocol_conformances),
-      SWIFT_SECTION_RANGE(swift5_type_metadata),
+    SWIFT_SECTION_RANGE(swift5_protocols),
+    SWIFT_SECTION_RANGE(swift5_protocol_conformances),
+    SWIFT_SECTION_RANGE(swift5_type_metadata),
 
-      SWIFT_SECTION_RANGE(swift5_typeref),
-      SWIFT_SECTION_RANGE(swift5_reflstr),
-      SWIFT_SECTION_RANGE(swift5_fieldmd),
-      SWIFT_SECTION_RANGE(swift5_assocty),
-      SWIFT_SECTION_RANGE(swift5_replace),
-      SWIFT_SECTION_RANGE(swift5_replac2),
-  };
+    SWIFT_SECTION_RANGE(swift5_typeref),
+    SWIFT_SECTION_RANGE(swift5_reflstr),
+    SWIFT_SECTION_RANGE(swift5_fieldmd),
+    SWIFT_SECTION_RANGE(swift5_assocty),
+    SWIFT_SECTION_RANGE(swift5_replace),
+    SWIFT_SECTION_RANGE(swift5_replac2),
+    SWIFT_SECTION_RANGE(swift5_builtin),
+    SWIFT_SECTION_RANGE(swift5_capture),
+};
 
 #undef SWIFT_SECTION_RANGE
 
+__asm__(".section \".note.swift_reflection_metadata\", \"aw\"");
+
+static
+__attribute__((__used__))
+__attribute__((__section__(".note.swift_reflection_metadata")))
+__attribute__((__aligned__(1)))
+char MagicString[] = SWIFT_REFLECTION_METADATA_ELF_NOTE_MAGIC_STRING;
+
+static
+__attribute__((__used__))
+__attribute__((__section__(".note.swift_reflection_metadata")))
+__attribute__((__aligned__(1)))
+const swift::MetadataSections *S = &sections;
```

Comment:
I don't think that this works.  The compiler is completely within its right to re-order the entries here.  So we can end up with an invalid section here.